data/tests/apg/modal-dialog-example.json

```diff
@@ -335,6 +335,13 @@
               "feature_assertion_id": "convey_name",
               "result": "pass",
               "notes": "The name and role were conveyed when entering the dialog for the first time."
+            },
+            {
+              "feature_id": "aria/aria-modal_attribute",
+              "feature_assertion_id": "convey_presence",
+              "applied_to": "aria/dialog_role",
+              "result": "fail",
```

Comment:
This is a bit tricky. I've been allowing the presence of `aria-modal` to be conveyed explicitly or implicitly.

* Explicitly: the screen reader actually says something like "modal" when the dialog opens.
* Implicitly: screen reader does NOT say something like "modal" the dialog opens, but instead, the user is unable to access content that was previously on the page or otherwise navigate outside of the dialog.

Doing some quick testing, it looks like JAWS conveys `aria-modal="true"` implicitly. I can't navigate outside of the dialog. There are some quirks... like I see headings outside of the modal listed in the headings list, but when I try to navigate to them I'm taken back to the dialog instead.

So, I'm tempted to say this passes.

Now that I've explained my thinking, I'm curious what you think. I shouldn't be the only person to make this decision.

Reply:
fwiw, i would tend to err on the side of the 'modal' functionality being supported is more important than whether 'modal' is announced or not... as that could also potentially come down to a verbosity setting or a decision by the developers of the screen reader that the 'modal' announcement was unnecessary or unwanted.

Reply:
I think "say it's modal" (explicit) and "restrict access to modal content only" (implicit) are 2 very distinct things. AT can definitely do one without the other.
In ARIA1.1, the lead statement of aria-modal is "Indicates whether an element is modal when displayed." I interpret that as a MUST / SHOULD statement - current a11ysupport expectation defines it as MUST. ✔️ 
Then ARIA spec follows with "Assistive technologies MAY limit navigation to the modal element's contents." My interpretation is that means the implicit statement above is a MAY expectation, but it's currently being bundled with a MUST expectation. 🤔 
It's more work (boo), and I'm absolutely happy to be overruled, but personally I'd split them to reflect the different expectation levels in the spec. 🤷

Reply:
as a general fwi, I have todo items (both assigned ARIA spec issues and other editorial clarifications I want to tackle) in regards to dialogs and aria-modal.  obviously any changes i propose would need to be accepted by the working group, but I do have ideas for updates to the role and this attribute that might help clarify things here.

Reply:
There is a lot of stuff to unpack here, some of which is unrelated to this specific topic. But I'm writing it all down to help me think it through.

Number 1: the spec says that the attribute "Indicates whether an element is modal when displayed". But how screen readers choose to do that can be variable... possibly conveying it explicitly or implicitly and the spec doesn't dictate how AT must do this.

Number 2: the spec says "assistive technologies SHOULD navigate to the element unless focus has explicitly been set elsewhere." - a11ysupport doesn't have an expectation for this defined. Should we test it?

Number 3: the spec says "Assistive technologies MAY limit navigation to the modal element's contents." a11ysupport has that as a SHOULD, mostly because the majority of tested AT at least try to implement it. I'm open to adjusting the strength. Should we change it to a "may"?

Number 4: the spec says "If focus moves to an element outside the modal element, assistive technologies SHOULD NOT limit navigation to the modal element." a11sypport does not currently test this. Should we?

Number 5: the spec says "When a modal element is displayed, authors MUST ensure the interface can be controlled using only descendants of the modal element." Since this is an author's responsibility, a11ysupport doesn't test this.

Number 6: the spec says "when a modal element is displayed, authors SHOULD mark all other contents as inert (such as "inert subtrees" in HTML) if the ability to do so exists in the host language." Since this is an author's responsibility (if the AT doesn't support it), a11ysupport doesn't test this.

@scottaohara brought up a good point about this potentially coming down to a verbosity setting or a decision by the developers of the screen reader that the 'modal' announcement was unnecessary or unwanted.

In my opinion, that's kind of the key. If the screen reader does decide to limit navigation to the modal content, then explicitly conveying the modal state could be redundant and unnecessarily verbose.

However, if a screen reader does NOT limit navigation to the modal content, then it's up to the author and an explicit aria-modal announcement may become more important... but could still be redundant/verbose if the author did in fact make the rest of the document inert, which they "SHOULD" (and arguably MUST from a WCAG perspective). 

So, because the spec doesn't say that assistive technologies MUST explicitly convey the modal state, and there is a rational reason for some screen readers to not do so... I still think the original wording of the expectation here is fine, and it can be satisfied either implicitly or explicitly. 

Does this sway you one way or the other @RavenAlly? 

Also, I'm curious what changes you are going to make @scottaohara!

Reply:
would mention here but would go off topic.  can discuss elsewhere.

Reply:
@RavenAlly I think this is what I'll do: I'll merge this and change the results to `pass`. If you still feel that we should revisit the expectations, can you open an issue and we can handle that separately?